### PR TITLE
Create xmake.lua for fltk

### PR DESCRIPTION
**Pull request by MoAlyousef**
* Before adding new features and new modules, please go to issues to submit the relevant feature description first.
* Write good commit messages and use the same coding conventions as the rest of the project.
* Please commit code to dev branch and we will merge into master branch in feature
* Ensure your edited codes with four spaces instead of TAB.

------

* 增加新特性和新模块之前，请先到issues提交相关特性说明，经过讨论评估确认后，再进行相应的代码提交，避免做无用工作。
* 编写友好可读的提交信息，并使用与工程代码相同的代码规范，代码请用4个空格字符代替tab缩进。
* 请提交代码到dev分支，如果通过，我们会在特定时间合并到master分支上。
* 为了规范化提交日志的格式，commit消息，不要用中文，请用英文描述。

### CODE REVIEW

**Comment by MoAlyousef** [comment]
Currently the CI for linux fails because of missing X11 libs.

**Comment by waruqi** [comment]
> Currently the CI for linux fails because of missing X11 libs.

You can add `add_deps("linxcursor")` or other x11 libs. 

Most of the x11 libraries have been included, if there are some, you can add them. Or continue to add them when I have time.

for example: https://github.com/xmake-io/xmake-repo/blob/master/packages/l/libx11/xmake.lua

**Comment by MoAlyousef** [comment]
I'll try that. I'll have to disable pango though since there isn't a package for it yet.

**Comment by waruqi** [comment]
> I'll try that. I'll have to disable pango though since there isn't a package for it yet.

If it is an optional dependency, you can disable them by default and enable them through customized configuration. for example 

https://github.com/xmake-io/xmake-repo/blob/c7806451e819b49817b0f9ae65757c7998cf4d96/packages/s/sfml/xmake.lua#L30-L34

If the pango package is also added in the future, we can enable it through `add_requires("fltk", {configs = {pango = true}})`.

**Comment by MoAlyousef** [comment]
It seems fontconfig fails to build on linux.

**Comment by waruqi** [comment]
> It seems fontconfig fails to build on linux.

I will fix it later.

**Comment by waruqi** [comment]
I have fixed fontconfig issue, you can rebase your commits from dev and try it again. https://github.com/xmake-io/xmake-repo/pull/159

File: packages/f/fltk/xmake.lua [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    if is_plat("windows", "mingw") then
+        add_syslinks("ws2_32", "comctl32", "gdi32", "oleaut32", "ole32", "uuid", "shell32", "advapi32", "comdlg32", "winspool", "user32", "kernel32", "odbc32")
+    elseif is_plat("macosx") then 
+        add_frameworks("Cocoa")
+    elseif is_plat("android") then
+        add_syslinks("android")
+        add_syslinks("dl")
+    else
+        add_configs("pango",   {description = "Use pango for font support", default = false, type = "boolean"})
+        add_configs("xft",     {description = "Use libXft for font support", default = false, type = "boolean"})
```
Comment (waruqi) [inline packages/f/fltk/xmake.lua:18]: Please move add_configs to outside.

```lua
if is_host("linux") then
        add_configs("pango",   {description = "Use pango for font support", default = false, type = "boolean"})
        add_configs("xft",     {description = "Use libXft for font support", default = false, type = "boolean"})
end
```

**Comment by MoAlyousef** [comment]
YES! 
All green.

**Comment by waruqi** [comment]
Ok, Thank you very much for your contribution. :+1:

**Comment by MoAlyousef** [comment]
Thank you for working on xmake and the tooling

**Pull request merged**